testsuite/tests/generated-parse-errors/errors.compilers.reference

```diff
@@ -1947,7 +1947,7 @@ Error: Syntax error
 File "implementation: TRUE DOT LBRACKET UIDENT RBRACKET LESSMINUS OBJECT END WHILE", line 1, characters 0-31:
 Error: Syntax error: strings are immutable, there is no assignment syntax for them.
 Hint: Mutable sequences of bytes are available in the Bytes module.
-Hint: Did you mean to use 'Bytes.set'?
+Hint: Did you mean to use Bytes.set?
```

Comment:
Haven't we discussed keeping a visible printing style for "dumb" terminals?

Reply:
Yes, we did. But I am not completely sure that having explicit quote marks is the best behavior for "dumb" terminals compared to printing directly the quoted text.

Reply:
If you do want a visual style, following the style of ocamldoc markup might be the best choice here: `Did you mean to use [Bytes.set]`.  But it seems a little inconsistent in the above example as just in the line above the `Bytes module` portion is unquoted.

It seems cleanest to me to just directly print the quoted text in a dumb terminal.

Reply:
I think that having some visible markup is useful in at least two situations:
- when printing user names, they can be harder to distinguish from normal text:
   + "The type above is required but not provided." vs.
   + "The type [above] is required but not provided"
- the visual cues can help scanning the message faster:
   + "The expression has type foo but an expression was expected of type bar" vs
   + "The expression has type [foo] but an expression was expected of type [bar]".

Reply:
I'm fine with visible markup too, but I think it should not be added only for dumb terminals.  Just using the ocamldoc-syntax everywhere seems cleanest, possibly with some augmentation of colour or other hints for fancier terminals.

Reply:
My opinion is that the styling for inline code should just provide a hint that the quoted contents is not part of the grammatical structure of the compiler message. I would thus prefer to use the lightest possible style for inline code. In particular, I would rather ensure that no one ever hesitate to add the inline code tag when editing compiler messages.

 In my opinion, a bold font weight fits well this constraint on most terminal, whereas ocamldoc markup is often too heavy.

Reply:
I have added the fallback mode for pure text terminal.